### ISSUE TITLE
[Version 8.0.0] Unable to connect to elasticsearch behind load balancer and nginx

### ISSUE THREAD

**Issue by nikhilkawtakwar** [elastic/elasticsearch-js#1625]
We were using npm package 7.15.0. we were able to connect to elasticsearch which is behind loadbalancer and nginx. However, I am getting invalid url issue when migrated to 8.0.0.

```
// client creation
import {Client} from '@elastic/elasticsearch'
const auth = {username: 'test', password: 'test'}

const esClient = new Client({
      node: <URL>, //https url
      auth,
      maxRetries: 5,
      requestTimeout: 60000,
    })

InvalidArgumentError: invalid url
    at Object.parseOrigin (/node_modules/undici/lib/core/util.js:83:11)
    at new Pool (/node_modules/undici/lib/pool.js:62:23)
    at new Connection (/@elastic/transport/lib/connection/UndiciConnection.js:101:21)
    at WeightedConnectionPool.createConnection (/node_modules/@elastic/transport/lib/pool/BaseConnectionPool.js:141:28)
    at WeightedConnectionPool.addConnection (/node_modules/@elastic/transport/lib/pool/BaseConnectionPool.js:164:59)
    at new Client (/@elastic/elasticsearch/lib/client.js:180:33)
```

Am i missing any config option here?

Environment
node version: v14.17.0
@elastic/elasticsearch version: 8.0.0
os: Mac

**Comment by delvedor** [comment]
Hello! Can you share the url? Or craft one that reproduces the issue?
Thanks!

**Comment by karol-gwaj** [comment]
hi, im getting similar error when using "8.2.0-patch.1" client version, 
the issue happens when proxying request to elasticsearch with url with path, like: https://example.com/elasticsearch/api
there is validation in parseOrigin function that expects the path to be '/' (and not '/elasticsearch/api')

**Comment by margaretjgu** [comment]
This has been fixed in 8.19.2 
https://github.com/elastic/elasticsearch-js/issues/3342
